File: LeetCode/0780. Reaching Points/solution.go
/**
 * 195 / 195 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 1.9 MB 
 */
 func reachingPoints(sx int, sy int, tx int, ty int) bool {
    for tx >= sx && ty >= sy {
        if tx == sx && ty == sy { return true }
        if tx > ty {
            if tx - sx < ty { break }
            tx -= ((tx - sx) / ty) * ty
        } else {
            if ty - sy < tx { break }
            ty -= ((ty - sy) / tx) * tx
        }
    }    
    return false
}

File: LeetCode/0780. Reaching Points/solution.py
"""
195 / 195 test cases passed.
Runtime: 44 ms
Memory Usage: 14.8 MB
"""
class Solution:
    def reachingPoints(self, sx: int, sy: int, tx: int, ty: int) -> bool:
        while tx >= sx and ty >= sy:
            if tx == sx and ty == sy:
                return True
            if tx > ty:
                tx -= max((tx - sx) // ty, 1) * ty
            else:
                ty -= max((ty - sy) // tx, 1) * tx
        return False
